### PR TITLE
Cookie consent

### LINKED ISSUE #428: gdpr compliance
## Is your feature request related to a problem? Please describe

Right now we don't have a cookie warning and consent for the trackable data. This does not follow EU regulations.

## Describe the solution you'd like

Please add a cookie consent solution, to be compliant, for example https://www.npmjs.com/package/vanilla-cookieconsent

## Additional context

We are currently using tracking with matomo and functional local storage for light/dark mode.

### PR DESCRIPTION
**Pull request by KirilCycle**
# The Pull Request is ready

- [x] fixes #428
- [ ] all actions are passing
- [x] only fixes a single issue

## Overview

I added simple modal to layout, with two options decline and acept

## Documentation-Website

- [ yes] mobile view is usable
- [ yes] desktop view is usable
- [ yes] all new texts are added to the translation files (at least the english one)

### CODE REVIEW

File: documentation-website/language/en.yml [hunk 1]
```diff
@@ -361,3 +361,9 @@ command:
     arg_4: "'iab load --language=en --taskId=test --threads=11 --repetitions=100' runs a load test with all options being optional."
     arg_5: "'iab bench --language=en --taskId=test --threads=11 --repetitions=100' runs a benchmark with all options being optional."
     arg_6: "'iab stress --language=en --taskId=test --threads=11 --repetitions=100 --duration=60' runs a stress test for the desired duration in seconds with all options being optional."
+
+cookie-consent:
+  title: 'We use cookies'
```
Comment (Idrinth) [inline documentation-website/language/en.yml:366]: double quotess please

File: documentation-website/language/en.yml [hunk 2]
```diff
@@ -361,3 +361,9 @@ command:
...
+  description: 'We use cookies to ensure you get the best experience on our website.'
+  accept: 'Accept'
+  decline: 'Decline'
```
Comment (Idrinth) [inline documentation-website/language/en.yml:369]: missing NewLine at EndOfFile

File: documentation-website/src/components/layout.tsx [hunk 3]
```diff
@@ -33,7 +32,7 @@ const Layout = ({
     </article>
     <Footer/>
     <ScrollToHashElement/>
-    {/* @ts-ignore  */}
+    {/*@ts-expect-error _paq can be null*/}
     {(!localStorage.getItem('consent-was-asked') && window?._paq) &&
```
Comment (Idrinth) [inline documentation-website/src/components/layout.tsx:36]: put the logic into the component please

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: documentation-website/src/components/cookie-consent.css [hunk 4]
```diff
@@ -0,0 +1,86 @@
+
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.css:1]: remove empty line please

File: documentation-website/src/tracking.ts [hunk 5]
```diff
@@ -13,6 +13,8 @@ if (window.location.host === 'idrinth-api-ben.ch') {
   window._paq = window._paq || [];
   // @ts-expect-error paq is a new property
   const _paq = window._paq;
+  _paq.push(['requireConsent']);
+
```
Comment (Idrinth) [inline documentation-website/src/tracking.ts:17]: empty line could be removed

File: documentation-website/src/tracking.ts [hunk 6]
```diff
@@ -35,4 +37,4 @@ if (window.location.host === 'idrinth-api-ben.ch') {
     'event': 'mtm.Start',
   },);
   make('js/container_Mf0xOjsp.js',);
-}
+}
```
Comment (Idrinth) [inline documentation-website/src/tracking.ts:40]: linebreak missing

File: documentation-website/src/components/window.ts [hunk 7]
```diff
@@ -5,6 +5,6 @@ interface Window {
   },
   Navigator: {
     language: string,
-  }
+  }
```
Comment (Idrinth) [inline documentation-website/src/components/window.ts:8]: no spaces at the end of line

File: documentation-website/src/components/cookie-consent.css [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  background-color: var(--dark-gray);
+}
+
+.cookie-consent-accept-button + .cookie-consent-decline-button {
+  margin-left: 10px;
+} 
+
+.dark-mode .cookie-consent-decline-button:hover {
+  opacity: 80%;
+}
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.css:86]: line  break missing

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: documentation-website/src/components/cookie-consent.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+} from './lang';
+
+const CookieConsent = () => {
+  const [
+    consentWasClosed,
+    setConsentWasClosed,
+  ] = useState<boolean>(false);
+
+  function handleConsent(accept: boolean) {
+    { /*@ts-expect-error _paq can be null*/ }
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.tsx:16]: why not use `?.`?

File: documentation-website/src/components/cookie-consent.tsx [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    window._paq.push([ accept ? 'rememberConsentGiven' : 'forgetConsentGiven', ]);
+    localStorage.setItem('consent-was-asked', 'yes');
+    setConsentWasClosed(true);
+  }
+
+  return (
+    <>
+      {/*@ts-expect-error _paq can be null*/}
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.tsx:24]: same as above

File: documentation-website/src/tracking.ts [hunk 11]
```diff
@@ -1,38 +1,38 @@
...
-  window._mtm = window._mtm || [];
-  // @ts-expect-error mtm is a new property
-  const _mtm = window._mtm;
-  _mtm.push({
-    'mtm.startTime': new Date().getTime(),
-    'event': 'mtm.Start',
-  },);
-  make('js/container_Mf0xOjsp.js',);
-}
+const FIRST = 0;
```
Comment (Idrinth) [inline documentation-website/src/tracking.ts:2]: this formatting is way off - similar to most files

**Review by Idrinth — CHANGES_REQUESTED** [review]
Please fix the eslint warnings as well

File: documentation-website/src/components/cookie-consent.tsx (hunk 9 above)
Comment (KirilCycle) [reply documentation-website/src/components/cookie-consent.tsx:16]: without comment  '?.' still gives an error

File: documentation-website/src/components/cookie-consent.css [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+  bottom: 20px;
+  z-index: 100;
+  left: 10px;
+  max-width: 410px;
+  box-shadow: 0 0 5px rgba(0, 0, 0, 10%);
+}
+
+/* Description styles */
+.cookie-consent-description {
+  text-align: left !important;
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.css:17]: no important

File: documentation-website/src/components/cookie-consent.css [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+.cookie-title {
+  margin: 0;
+  text-align: left;
+  font-size: 18px;
+}
+
+.cookie-description {
+  font-size: 14px;
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.css:30]: why the different font-size?

File: documentation-website/src/components/cookie-consent.css [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+  color: #fff;
+}
+
+.cookie-consent-accept-button:hover {
+  background-color: var(--light-green);
+}
+
+.cookie-consent-decline-button {
+  background-color: var(--light-gray);
+  color:  --black;
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.css:60]: var missing

File: documentation-website/src/components/cookie-consent.css [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+.dark-mode .cookie-description {
+  color: var(--light-gray);
+}
+
+.dark-mode .cookie-consent-decline-button { 
+  color: var(--white);
+  background-color: var(--dark-gray);
+}
+
+.cookie-consent-accept-button + .cookie-consent-decline-button {
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.css:79]: Why is the neighbouring check required?

File: documentation-website/src/components/cookie-consent.tsx [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+    const action = accept ? 'rememberConsentGiven' : 'forgetConsentGiven';
+    { /*@ts-expect-error _paq can be null*/ }
+    window?._paq.push([ action, ],);
+    localStorage.setItem('consent-was-asked', 'yes',);
+    setConsentWasClosed(true,);
+  };
+
+  return (
+    <>
+      {! consentWasClosed &&
```
Comment (Idrinth) [inline documentation-website/src/components/cookie-consent.tsx:25]: extract that logic into an if with a different return please

**Pull request merged**